src/LauncherManager.vala

```diff
@@ -145,18 +152,45 @@
         }
 
         if (reposition) {
-            reposition_launchers ();
+            resize_animation.easing = EASE_OUT_BACK;
+            resize_animation.duration = Granite.TRANSITION_DURATION_OPEN;
+            resize_animation.value_from = get_width ();
+            resize_animation.value_to = launchers.length () * get_launcher_size ();
+            resize_animation.play ();
+
+            resize_animation.done.connect (() => {
```

Comment:
Also I think this has to be disconnected otherwise if you have apps that were added later and then another one is added all of the apps that were added later are playing the reveal animation again

Reply:
Yup you're right haha. Good catch